### DISCUSSION TITLE
Checkbox design

### DISCUSSION THREAD

**Discussion by half5cat** [carbon-design-system/carbon#9775] [Help] [answered]
I am currently updating my Figma Library for Carbon and have trouble understanding what is actually the correct way of implementing the checkboxes. In the documentation it looks like it uses the checkbox icons and that's how I did it until now:
https://www.carbondesignsystem.com/components/checkbox/usage#states 

However, the implementation is showing something quite different:
https://react.carbondesignsystem.com/?path=/story/components-checkbox--checkbox 

It looks like the size is different and there are no rounded corners.

I would assume that the implementation is correct. If I wanted to rebuild this, which icon in which size should i use for the checked state and the indeterminate state?

**Comment by jnm2377** [comment, marked as answer]
Hey @half5cat the reason you're seeing it look different is because the implementation doesn't use an icon. We are using pseudoelements to make the checked look. This is likely something that never changed from older versions, I'm unsure why an icon was not implemented. I found an example of a checkbox component that uses custom svgs: https://codepen.io/rolchau/pen/rlAtq/

I think implementing something like this would be a breaking change for us, though, and ,in general, low priority as we're focusing on a11y issues for our next major version right now.

**Reply by half5cat** [reply]
That makes sense. I think the documentation should be updated accordingly.
